src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -115,15 +126,31 @@ public override void OnEventWritten(string name, object? payload)
                                 case CommandType.StoredProcedure:
                                     if (this.options.SetDbStatementForStoredProcedure)
```

Comment:
Before making this package stable, we should reconsider this feature flag name.

same for `SetDbStatementForText`. It can be separate issue.

Reply:
Yes, all of the options will need to be reviewed before we ship stable. I lean toward only having one option for turning collection of `db.query.text` on/off. These options are also related to the sanitization I'll implement with #2221. Rather than just on/off, these options may need to be something like: off, on w/ sanitization, and on w/o sanitization.